x-pack/plugins/monitoring/public/views/base_controller.js

```diff
@@ -96,12 +96,24 @@ export class MonitoringViewBaseController {
       timefilter.enableAutoRefreshSelector();
     }
 
+    this.updateDataPromise = null;
+    this.updatingData = false;
     this.updateData = () => {
+      if (this.updatingData) {
```

Comment:
We could simplify this and remove `this.updatingData` altogether by simply checking `this.updateDataPromise`, then just set `this.updateDataPromise = null;`, which we should probably do regardless, instead of `this.updatingData = false;`.